src/queries.ts

```diff
@@ -0,0 +1,128 @@
+import { GQLEdgeInterface, GQLTransactionsResultInterface } from './gql_types';
+import fetch from 'node-fetch';
+
+const GQL_URL = 'https://arweave.net/graphql';
+const ITEMS_PER_REQUEST = 100;
+const VALID_APP_NAMES = ['ArDrive-Web', 'ArDrive-CLI', 'ArDrive-Sync'] as const;
+
+interface Query {
+	query: string;
+}
+
+export interface StakedPSTHolders {
+	[address: string]: number;
+}
+
+export async function getWaleltsEligibleForStreak(): Promise<StakedPSTHolders> {
+	return getStakedPSTHolders()
+		.then((result) => Object.entries(result))
+		.then((entries) => entries.filter((data) => data[1] >= 200))
+		.then((entries) => Object.fromEntries(entries));
+}
+
+async function getStakedPSTHolders(): Promise<StakedPSTHolders> {
+	// https://v2.cache.verto.exchange/-8A6RexFkpfWwuyVO98wzSFZh0d6VJuI-buTJvlwOJQ
+	// const result = await vertoClient.getExchangeDetails('ARDRIVE');
+	const blockHeightRequest = await fetch('https://arweave.net/height');
+	const blockHeight = +(await (await blockHeightRequest.blob()).text());
+	const result = await fetch('https://v2.cache.verto.exchange/-8A6RexFkpfWwuyVO98wzSFZh0d6VJuI-buTJvlwOJQ');
+	const holders = await result.json();
+	const vault = holders.state.vault;
```

Comment:
```suggestion
	const { state: { vault } } = await result.json();
```

Reply:
I might want to create a type for that :opionion:

Reply:
[e6da8e6](https://github.com/ardriveapp/inferno-backend/pull/2/commits/e6da8e6c1dbf9c46e518fa9ad42b77aad6fe8c76)